packages/auth/src/api/authentication/idp.test.ts

```diff
@@ -87,4 +87,35 @@ describe('api/authentication/signInWithIdp', () => {
     );
     expect(mock.calls[0].request).to.eql(request);
   });
+  it('should handle user disabled errors and propagate email info', async () => {
+    const mock = mockEndpoint(
+      Endpoint.SIGN_IN_WITH_IDP,
+      {
+        email: "test123@gmail.com",
+        error: {
+          code: 400,
+          message: ServerError.USER_DISABLED,
+          errors: [
+            {
+              message: ServerError.USER_DISABLED
+            }
+          ]
+        }
+      },
+      400
+    );
+
+    let error: FirebaseError | null = null;
+    try {
+    await signInWithIdp(auth, request);
+    } catch (e) {
+      error = e;
+    }
+    expect(error!.customData!.email).to.eq("test123@gmail.com");
+    expect(error!.code).to.eq(
+      'auth/user-disabled'
+    );
```

Comment:
You can use chai-as-promised to put all of this error testing logic into one assertion phrase. From ap/index.test.ts for example:
https://github.com/firebase/firebase-js-sdk/blob/b6196d28ce6117e17dbe87cfd554d568012ef31d/packages/auth/src/api/index.test.ts#L178-L183

i.e.

```js
await expect(signInWithIdp(auth, request)).to.be.rejectedWith('auth/user-disabled')
    .eventually.with.deep.property('customData', {email: 'test123@gmail.com'});
```

Reply:
Done.